File: 1. Two Sum/Solution.java
import java.util.HashMap;

public class Solution {
	public int [] twoSum(int [] nums, int target) {
		int [] ret = new int [2];
		HashMap<Integer, Integer> hash = new HashMap<>();
		for (int i = 0; i < nums.length; i++) {
			if (hash.get(target - nums[i]) != null) {
				ret[0] = i;
				ret[1] = hash.get(target - nums[i]);
				return ret;
			} 
			hash.put(nums[i], i);
		}
		return ret;
	}
}

File: 1. Two Sum/Solution.py
class Solution:
	def twoSum(self, nums: list[int], target: int) -> list[int]:
		hash = dict();
		for i in range(len(nums)):
			if hash.get(target - nums[i]) != None:
				return [i, hash[target - nums[i]]]
			hash[nums[i]] = i



if __name__ == '__main__':
	sl = Solution()
	nums = [2, 7, 11, 15]
	res = sl.twoSum(nums, 9)
	print(res)